--- a/src/d_main.cpp
+++ b/src/d_main.cpp
@@ -110,6 +110,7 @@
 #include "s_music.h"
 #include "filereadermusicinterface.h"
 #include "r_videoscale.h"
+#include "archipelago/archipelago_client.h"
 
 EXTERN_CVAR(Bool, hud_althud)
 EXTERN_CVAR(Int, vr_mode)
@@ -3607,6 +3608,9 @@ static int D_DoomMain_Internal (void)
 	}
 	
 	D_DoomInit();
+	
+	// Initialize Archipelago client
+	Archipelago::AP_Init();
 
 	// [RH] Make sure zdoom.pk3 is always loaded,
 	// as it contains magic stuff we need.
@@ -3812,6 +3816,8 @@ int GameMain()
 	catch (const std::exception &error)
 	{
 		I_ShowFatalError(error.what());
 		ret = -1;
 	}
+	
+	Archipelago::AP_Shutdown();
 	// Unless something really bad happened, the game should only exit through this single point in the code.
 	// No more 'exit', please.