--- a/src/d_net.cpp
+++ b/src/d_net.cpp
@@ -73,6 +73,7 @@
 #include "serializer.h"
 #include "s_music.h"
 #include "d_buttons.h"
+#include "archipelago/archipelago_client.h"
 
 extern FRandom pr_spawnmobj;
 
@@ -2922,6 +2923,13 @@ void Net_DoCommand (int type, uint8_t **stream, int player)
 
 void Net_SkipCommand (int type, uint8_t **stream)
 {
 	size_t skip = NetCmds[type]->MessageSize(stream);
 	*stream += skip;
 }
+
+// Called from the main game loop
+void Net_ProcessArchipelago()
+{
+	if (Archipelago::g_archipelago && Archipelago::g_archipelago->IsConnected()) {
+		Archipelago::g_archipelago->ProcessMessages();
+	}
+}